proxywasm/proxytest/http.go

```diff
@@ -406,9 +407,15 @@ func (h *httpHostEmulator) CallOnRequestBody(contextID uint32, body []byte, endO
 		log.Fatalf("invalid context id: %d", contextID)
 	}
 
-	cs.requestBody = append(cs.requestBody, body...)
+	cs.requestBody = append(cs.requestBodyBuffer, body...)
 	cs.action = internal.ProxyOnRequestBody(contextID,
 		len(body), endOfStream)
+	if cs.action == types.ActionPause {
+		// Buffering requested
+		cs.requestBodyBuffer = cs.requestBody
```

Comment:
Fixed test I think - change ended up being a bit more complicated than I expected since I realized that the body is also worked on by the plugin. If we reset the body completely when continuing, then that would be like not sending a body to upstream. So I think it means we need a separate buffer, for consecutive calls - continue only clears the buffer, but not the body created by the plugin.